pkgs/development/java-modules/maven-minimal.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv, pkgs, maven }:
+
+with stdenv.lib;
+with pkgs.javaPackages;
+
+let
+  collections = import ./collections.nix { inherit pkgs; };
+  fetchMaven = pkgs.callPackage ./m2install.nix { };
+  plugins = import ./mavenPlugins.nix { inherit stdenv pkgs maven; };
+  poms = import ./poms.nix { inherit fetchMaven; };
+in rec {
+  # Maven needs all of these to function
+  mavenMinimal = flatten
+    collections.mavenLibs_2_0_6
+    ++ collections.mavenLibs_2_0_9
+    ++ collections.mavenLibs_2_2_1
+    ++ [
+      classworlds_1_1_alpha2
+      classworlds_1_1
+      commonsCli_1_0
+      commonsLang_2_1
+      commonsLang3_3_1
+      commonsLoggingApi_1_1
+      findbugsJsr305_2_0_1
+      googleCollections_1_0
+      junit_3_8_1
```

Comment:
Agreed, you could strip certain dependencies if you choose to skip tests. However, that did not seem worth the effort right now. Most builds from source should want to not skip tests :)